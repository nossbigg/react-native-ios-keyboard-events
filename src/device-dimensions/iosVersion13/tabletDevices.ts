import { IDeviceModel } from "../deviceDimensions";

const tabletDevices: IDeviceModel[] = [
  // For iPad Air, iPad Air 2, iPad (5th generation), iPad (6th generation), and iPad Pro (9.7-inch)
  {
    model: "iPad Air",
    isTablet: true,
    screenDimensions: new Set([1024, 768]),
    keyboardDimensions: {
      landscape: {
        docked: 408,
        split: 271,
        minimized: 55,
      },
      portrait: {
        docked: 320,
        split: 271,
        minimized: 55,
      },
    },
  },
  {
    model: "iPad Pro (10.5-inch)",
    isTablet: true,
    screenDimensions: new Set([1112, 834]),
    keyboardDimensions: {
      landscape: {
        docked: 408,
        split: 271,
        minimized: 55,
      },
      portrait: {
        docked: 320,
        split: 271,
        minimized: 55,
      },
    },
  },
  {
    model: "iPad Pro (11-inch)",
    isTablet: true,
    screenDimensions: new Set([1194, 834]),
    keyboardDimensions: {
      landscape: {
        docked: 428,
        minimized: 69,
      },
      portrait: {
        docked: 340,
        minimized: 69,
      },
    },
  },
  // For iPad Pro (12.9-inch), iPad Pro (12.9-inch) (2nd generation), and iPad Pro (12.9-inch) (3rd generation)
  {
    model: "iPad Pro (12.9-inch)",
    isTablet: true,
    screenDimensions: new Set([1366, 1024]),
    keyboardDimensions: {
      landscape: {
        docked: 478,
        minimized: 55,
      },
      portrait: {
        docked: 383,
        minimized: 55,
      },
    },
  },
];

export default tabletDevices;
