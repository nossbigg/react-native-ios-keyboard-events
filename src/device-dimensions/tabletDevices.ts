import { IDeviceInformation } from "./deviceDimensions";

const tabletDevices: IDeviceInformation[] = [
  // For iPad Air, iPad Air 2, iPad (5th generation), iPad (6th generation), and iPad Pro (9.7-inch)
  {
    model: "iPad Air",
    isTablet: true,
    screenDimensions: new Set([1024, 768]),
    keyboardDimensions: {
      landscape: {
        docked: 398,
        split: 271,
        minimized: 55,
      },
      portrait: {
        docked: 313,
        split: 271,
        minimized: 55,
      },
    },
  },
  {
    model: "iPad Pro (10.5-inch)",
    isTablet: true,
    screenDimensions: new Set([1112, 834]),
    keyboardDimensions: {
      landscape: {
        docked: 398,
        split: 271,
        minimized: 55,
      },
      portrait: {
        docked: 313,
        split: 271,
        minimized: 55,
      },
    },
  },
  {
    model: "iPad Pro (11-inch)",
    isTablet: true,
    screenDimensions: new Set([1194, 834]),
    keyboardDimensions: {
      landscape: {
        docked: 418,
        minimized: 69,
      },
      portrait: {
        docked: 333,
        minimized: 69,
      },
    },
  },
  // For iPad Pro (12.9-inch), iPad Pro (12.9-inch) (2nd generation), and iPad Pro (12.9-inch) (3rd generation)
  {
    model: "iPad Pro (12.9-inch)",
    isTablet: true,
    screenDimensions: new Set([1366, 1024]),
    keyboardDimensions: {
      landscape: {
        docked: 471,
        minimized: 55,
      },
      portrait: {
        docked: 378,
        minimized: 55,
      },
    },
  },
];

export default tabletDevices;
