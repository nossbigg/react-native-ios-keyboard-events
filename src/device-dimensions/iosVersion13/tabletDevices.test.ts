import tabletDevices from "./tabletDevices";

describe("tabletDevices", () => {
  test("matches snapshot", () => {
    expect(tabletDevices).toMatchSnapshot();
  });
});
