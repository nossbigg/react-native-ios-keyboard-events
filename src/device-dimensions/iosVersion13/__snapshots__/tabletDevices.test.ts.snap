// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tabletDevices matches snapshot 1`] = `
Array [
  Object {
    "isTablet": true,
    "keyboardDimensions": Object {
      "landscape": Object {
        "docked": 408,
        "minimized": 55,
        "split": 271,
      },
      "portrait": Object {
        "docked": 320,
        "minimized": 55,
        "split": 271,
      },
    },
    "model": "iPad Air",
    "screenDimensions": Set {
      1024,
      768,
    },
  },
  Object {
    "isTablet": true,
    "keyboardDimensions": Object {
      "landscape": Object {
        "docked": 408,
        "minimized": 55,
        "split": 271,
      },
      "portrait": Object {
        "docked": 320,
        "minimized": 55,
        "split": 271,
      },
    },
    "model": "iPad Pro (10.5-inch)",
    "screenDimensions": Set {
      1112,
      834,
    },
  },
  Object {
    "isTablet": true,
    "keyboardDimensions": Object {
      "landscape": Object {
        "docked": 428,
        "minimized": 69,
      },
      "portrait": Object {
        "docked": 340,
        "minimized": 69,
      },
    },
    "model": "iPad Pro (11-inch)",
    "screenDimensions": Set {
      1194,
      834,
    },
  },
  Object {
    "isTablet": true,
    "keyboardDimensions": Object {
      "landscape": Object {
        "docked": 478,
        "minimized": 55,
      },
      "portrait": Object {
        "docked": 383,
        "minimized": 55,
      },
    },
    "model": "iPad Pro (12.9-inch)",
    "screenDimensions": Set {
      1366,
      1024,
    },
  },
]
`;
